spec/models/manageiq/providers/ibm_cloud/vpc/cloud_tools_spec.rb

```diff
@@ -219,16 +219,55 @@ def create_bearer_info(valid_expire_time: true, set_api_key: true)
   end
 
   it 'Raises error with non-existent method.' do
-    vpc = described_class.new(:api_key => api_key).vpc(:region => 'us-east')
     bad_method = :list_instancess
 
     expect { vpc.request(bad_method) }.to raise_error(StandardError, 'IbmVpc::VpcV1 does not contain a method list_instancess')
     expect { vpc.collection(bad_method).first }.to raise_error(StandardError, 'IbmVpc::VpcV1 does not contain a method list_instancess')
   end
 
   it 'Raises error in collection when method is not a list.' do
-    vpc = described_class.new(:api_key => api_key).vpc(:region => 'us-east')
-
     expect { vpc.collection(:get_instance) }.to raise_error(StandardError, 'Provided call_back get_instance does not start with list. This method is for paginating list methods.')
   end
+
+  it 'Instances is callable' do
+    instances = vpc.instances
+    expect(instances).to be_a(ManageIQ::Providers::IbmCloud::CloudTools::VpcSdk::Instances)
+    expect(instances).to be_a(ManageIQ::Providers::IbmCloud::CloudTools::Sdk::Leaf)
+    expect(instances).to respond_to(:all)
+    expect(instances.all(:limit => 1)).to be_a(Enumerator)
+    expect(instances.all.first).to be_a(Hash)
+  end
+
+  it 'Instance is callable' do
+    expect(vpc.instances).to respond_to(:instance).with(1).argument
+
+    # Request a list of instances and limit to 1 result. Test each instance to see if it is running.
+    instance_hash = nil
+    vpc.instances.all(:limit => 1).each do |inst|
+      instance_hash = inst if inst[:status].downcase == 'running'
+      break if instance_hash.nil? == false
+    end
```

Comment:
It looks like you want to set the `instance_hash` var to the first `inst` where the `status` is running, this can be done a little simpler:
`instance_hash = vpc.instances.all(:limit => 1).detect { |inst| inst[:status].downcase == 'running' }`

Reply:
I tried detect and it failed:
```
  1) ManageIQ::Providers::IbmCloud::CloudTool Instance is callable
     Failure/Error: expect(instance_hash).not_to be_nil, 'Unable to find a running VM in VPC Cloud.'
       Unable to find a running VM in VPC Cloud.
     # ./spec/models/manageiq/providers/ibm_cloud/vpc/cloud_tools_spec.rb:247:in `block (2 levels) in <top (required)>'      
```

Reverting to my original code passes.

```
** Using session_store: ActionDispatch::Session::MemoryStore
** ManageIQ master, codename: Morphy

Randomized with seed 4621
.......................

Finished in 0.95595 seconds (files took 23.58 seconds to load)
23 examples, 0 failures
```

Reply:
```
$ git diff
diff --git a/spec/models/manageiq/providers/ibm_cloud/vpc/cloud_tools_spec.rb b/spec/models/manageiq/providers/ibm_cloud/vpc/cloud_tools_spec.rb
index 010ffd2a..9d59adec 100644
--- a/spec/models/manageiq/providers/ibm_cloud/vpc/cloud_tools_spec.rb
+++ b/spec/models/manageiq/providers/ibm_cloud/vpc/cloud_tools_spec.rb
@@ -242,11 +242,7 @@ describe ManageIQ::Providers::IbmCloud::CloudTool, :vcr do
     expect(vpc.instances).to respond_to(:instance).with(1).argument
 
     # Request a list of instances and limit to 1 result. Test each instance to see if it is running.
-    instance_hash = nil
-    vpc.instances.all(:limit => 1).each do |inst|
-      instance_hash = inst if inst[:status].downcase == 'running'
-      break if instance_hash.nil? == false
-    end
+    instance_hash = vpc.instances.all(:limit => 1).detect { |inst| inst[:status].downcase == 'running' }
     expect(instance_hash).not_to be_nil, 'Unable to find a running VM in VPC Cloud.'
```

```
$ rspec spec/models/manageiq/providers/ibm_cloud/vpc/cloud_tools_spec.rb
** Using session_store: ActionDispatch::Session::MemoryStore
** ManageIQ master, codename: Morphy
The version of PostgreSQL being connected to is incompatible with ManageIQ (10 required)
The version of PostgreSQL being connected to is incompatible with ManageIQ (10 required)

Randomized with seed 18872
.......................

Finished in 0.26079 seconds (files took 6.29 seconds to load)
23 examples, 0 failures

Randomized with seed 18872
```

Reply:
Bah, I missed the assignment to `instance_hash`. Works now.